packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts

```diff
@@ -133,84 +170,92 @@ export function getOrgConnectionInfo({
   return { orgId, autoAccept };
 }
 
-export async function createNewUserConnectToOrgIfExists({
-  usernameOrEmail,
+export async function createNewUsersConnectToOrgIfExists({
+  usernamesOrEmails,
   input,
   parentId,
   autoAcceptEmailDomain,
-  connectionInfo,
+  connectionInfoMap,
 }: {
-  usernameOrEmail: string;
+  usernamesOrEmails: string[];
   input: InviteMemberOptions["input"];
   parentId?: number | null;
   autoAcceptEmailDomain?: string;
-  connectionInfo: ReturnType<typeof getOrgConnectionInfo>;
+  connectionInfoMap: Record<string, ReturnType<typeof getOrgConnectionInfo>>;
 }) {
-  const { orgId, autoAccept } = connectionInfo;
-
-  const [emailUser, emailDomain] = usernameOrEmail.split("@");
-  const username =
-    emailDomain === autoAcceptEmailDomain
-      ? slugify(emailUser)
-      : slugify(`${emailUser}-${emailDomain.split(".")[0]}`);
-
-  const createdUser = await prisma.user.create({
-    data: {
-      username,
-      email: usernameOrEmail,
-      verified: true,
-      invitedTo: input.teamId,
-      organizationId: orgId || null, // If the user is invited to a child team, they are automatically added to the parent org
-      teams: {
-        create: {
-          teamId: input.teamId,
-          role: input.role as MembershipRole,
-          accepted: autoAccept, // If the user is invited to a child team, they are automatically accepted
+  await prisma.$transaction(async (tx) => {
+    for (let index = 0; index < usernamesOrEmails.length; index++) {
+      const usernameOrEmail = usernamesOrEmails[index];
+      const { orgId, autoAccept } = connectionInfoMap[usernameOrEmail];
+      const [emailUser, emailDomain] = usernameOrEmail.split("@");
+      const username =
+        emailDomain === autoAcceptEmailDomain
+          ? slugify(emailUser)
+          : slugify(`${emailUser}-${emailDomain.split(".")[0]}`);
+
+      const createdUser = await tx.user.create({
+        data: {
+          username,
+          email: usernameOrEmail,
+          verified: true,
+          invitedTo: input.teamId,
+          organizationId: orgId || null, // If the user is invited to a child team, they are automatically added to the parent org
+          teams: {
+            create: {
+              teamId: input.teamId,
+              role: input.role as MembershipRole,
+              accepted: autoAccept, // If the user is invited to a child team, they are automatically accepted
+            },
+          },
```

Comment:
we could conditionally create the membership to the Organization here instead of doing it in the  `if statement` below, but it would not have any impact since prisma would still make different sql queries